### PR TITLE
Add http generator tests

### PR DESCRIPTION
**Pull request by NadhiyaNashath**
## Purpose
The existing test suite for http code generator had insufficient coverage and not every test suites can be reused for the refactored code. Tests had been added for the new architecture

## Summary

This pull request enhances test coverage for the HTTP code generator following an architectural refactor. The changes include eight new test classes covering both unit and integration testing, configuration updates to support the test suite, and multiple test fixture files.

### Build and Configuration Updates
- Updated `build.gradle` to remove the `application` plugin and add TestNG test execution configuration pointing to the test suite in `testng.xml`
- Extended `checkstyle` task configuration to validate test code in addition to main code
- Added `log4j2.xml` configuration to establish logging for the test suite

### Test Coverage Additions
Eight comprehensive test classes were added:
- **Integration Testing**: `HttpCodeGeneratorIntegrationTest` validates the complete code generation pipeline across multiple scenarios (body identifier, header identifier, inline schemas) and error cases (null output path, missing event identifier extension)
- **Event Identifier Extraction**: `EventIdentifierExtractorTest` covers extraction logic for both body and header-based identifiers, including keyword escaping and error handling for missing/invalid configurations
- **Schema and Service Extraction**: `SchemaExtractorTest` and `ServiceTypeExtractorTest` validate schema and service type extraction from AsyncAPI specifications, with coverage for empty/missing components and publish-only operations
- **Code Generation**: `DispatcherGeneratorTest`, `ListenerGeneratorTest`, and `ServiceTypesGeneratorTest` verify source code generation for dispatcher services, listeners, and service type definitions, including validation of error cases
- **Utilities**: `CodegenUtilsTest` provides comprehensive coverage for identifier escaping, name sanitization, reference resolution, and naming conventions

### Test Fixtures
Five AsyncAPI specification fixtures were added to support testing across different scenarios:
- `github.json` - Multi-channel specification with multiple event types using body identifier mapping
- `stripe.json` - Header-based event identifier specification with `oneOf` payload mappings
- `sendgrid_minimal.json` - Minimal specification with inline schema payloads
- `no_operations.json` - Specification with no subscribe operations (edge case)
- `missing_event_identifier.json` - Specification intentionally missing the event identifier extension (error case)

### TestNG Configuration
Added `testng.xml` to define a test suite encompassing all eight test classes, enabling organized test execution across the HTTP code generator components.

## Outcomes
- Comprehensive test coverage for HTTP code generator functionality
- Test suite alignment with refactored architecture
- Coverage of normal operations, edge cases, and error scenarios
- Reusable test fixtures supporting multiple test contexts

### CODE REVIEW

File: asyncapi-generator/src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (gayaldassanayake) [inline asyncapi-generator/src/main/resources/log4j2.xml]: What is the reason for this log4j file?

**Comment by gayaldassanayake** [comment]
@coderabbitai please review

File: asyncapi-generator/build.gradle [hunk 2]
```diff
@@ -18,12 +18,9 @@
 
 plugins {
     id 'java'
-    id 'application'
     id 'checkstyle'
 }
 
-description = "Ballerina - AsyncAPI Generator"
```
Comment (gayaldassanayake) [inline asyncapi-generator/build.gradle:25]: Any reason for removing this? IIRC, this is there in other repos as well. Check openapi tool for reference and verify.

Comment (ayeshLK) [reply asyncapi-generator/build.gradle:25]: +1 to keep it.

Comment (NadhiyaNashath) [reply asyncapi-generator/build.gradle:25]: Added it
